AStarAI/src/main/java/com/github/sef24sp4/astarai/caching/DumbCache.java

```diff
@@ -0,0 +1,80 @@
+package com.github.sef24sp4.astarai.caching;
+
+import com.github.sef24sp4.astarai.AStar;
+import com.github.sef24sp4.astarai.Node;
+import com.github.sef24sp4.common.ai.map.Map;
+import com.github.sef24sp4.common.entities.ICollidableEntity;
+import com.github.sef24sp4.common.vector.IVector;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+public class DumbCache implements IPathCaching {
+
+	private final ICollidableEntity entity;
+	private AStar aStarSession;
+	private Map map;
+	private Node entityNode;
+	private Node targetNode;
+	private List<Node> pathList = new LinkedList<>();
+
+	private long lastTimeCalled;
+
+	public DumbCache(final ICollidableEntity entity) {
+		this.entity = entity;
+	}
+
+	@Override
+	public boolean flush() {
+		this.aStarSession = null;
+		return false;
+	}
+
+	@Override
+	public IVector getNextCoordinates(ICollidableEntity collidableEntity, final IVector targetCoordinate) {
+
+		if (this.isCachedPathValid(targetCoordinate)) { //if cache is valid, return (optional Node) or null
+			return this.getNextNode().map(Node::getCoordinates).orElse(null);
+		}
+
+		this.map.getNodeContaining(this.entity.getCoordinates()).ifPresent(node -> {
+			this.entityNode = new Node(this.entity.getCoordinates(), node);
+		});
+		this.map.getNodeContaining(collidableEntity.getCoordinates()).ifPresent(node -> {
+			this.targetNode = new Node(targetCoordinate, node);
+		});
+
+		//if no cache : calc new route and take next node from pathlist.
+		this.aStarSession = new AStar(this.entityNode, this.targetNode);
```

Comment:
`NullPointerException` is likely to be thrown, as `entityNode` and `targetNode` is not guaranteed to be initialised.

Reply:
<img width="599" alt="image" src="https://github.com/SE-F24-SP4/semesterproject-4/assets/116833415/547c906c-f0fa-4c26-a7d6-b3e9846b1343">
Would this solve the problem, or how do I make sure not to get a NullpointerExecption?

Reply:
No you are merely throwing an `IllegalArgumentException` instead of a `NullPointerException`. 
You need to restructure the code to take into the account `this.map.getNodeContaining(collidableEntity.getCoordinates())` possibly returning an empty `Optional`